test/widgets/action_sheet_test.dart

```diff
@@ -0,0 +1,229 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/api/route/messages.dart';
+import 'package:zulip/model/compose.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+import 'package:zulip/widgets/compose_box.dart';
+import 'package:zulip/widgets/content.dart';
+import 'package:zulip/widgets/message_list.dart';
+import 'package:zulip/widgets/store.dart';
+import '../api/fake_api.dart';
+
+import '../example_data.dart' as eg;
+import '../model/binding.dart';
+import '../model/test_store.dart';
+import 'compose_box_checks.dart';
+import 'dialog_checks.dart';
+
+void main() {
+  group('QuoteAndReplyButton', () {
+    TestDataBinding.ensureInitialized();
+
+    /// Simulates loading a [MessageListPage] and long-pressing on [message].
+    Future<void> setupToMessageActionSheet(WidgetTester tester, {
+      required Message message,
+      required Narrow narrow,
+    }) async {
+      addTearDown(TestDataBinding.instance.reset);
+
+      await TestDataBinding.instance.globalStore.add(eg.selfAccount, eg.initialSnapshot);
+      final store = await TestDataBinding.instance.globalStore.perAccount(eg.selfAccount.id);
+      store.addUser(eg.user(userId: message.senderId));
+      if (message is StreamMessage) {
+        store.addStream(eg.stream(streamId: message.streamId));
+      }
+      final connection = store.connection as FakeApiConnection;
+
+      // prepare message list data
+      connection.prepare(json: GetMessagesResult(
+        anchor: message.id,
+        foundNewest: true,
+        foundOldest: true,
+        foundAnchor: true,
+        historyLimited: false,
+        messages: [message],
+      ).toJson());
+
+      await tester.pumpWidget(
+        MaterialApp(
+          home: GlobalStoreWidget(
+            child: PerAccountStoreWidget(
+              accountId: eg.selfAccount.id,
+              child: MessageListPage(narrow: narrow)))));
+
+      // global store, per-account store, and message list get loaded
+      await tester.pumpAndSettle();
+
+      // request the message action sheet
+      final messageContent = tester.widget<MessageContent>(
+        find.byWidgetPredicate((Widget widget) => widget is MessageContent && widget.message.id == message.id));
+      await tester.longPress(find.byWidget(messageContent));
```

Comment:
> The widget `messageContent` is used only for passing back to a `find.byWidget`. So the original finder could be used directly, instead of going through `tester.widget`.

I think my intent here was to error (in the `tester.widget` call) before accidentally passing a finder to `tester.longPress` that would find multiple widgets, since it's not clear what `tester.longPress` would naturally do with such a finder.

…Upon reading the code, I've discovered that it throws a descriptive error, so we don't need our own defensive code. 😅

> Also we've constructed things so there's only one message in sight, so the message-ID check seems redundant. […]

I think in particular it helps that that construction of the message list (with just one message in sight) is done locally in this same function. So this seems fine.